src/internal/utils/convertSnakeToCamelCase.ts

```diff
@@ -0,0 +1,31 @@
+/**
+ * Converts snake_case keys to camelCase in an object or array of objects.
+ * @param {T} obj - The object, array, or string to convert. (required)
+ * @returns {T} The converted object, array, or string.
+ */
+export function convertSnakeToCamelCase<T>(obj: T): T {
+  if (typeof obj === 'string') {
+    return obj as T;
+  }
+
+  if (Array.isArray(obj)) {
+    return obj.map((item) => convertSnakeToCamelCase(item)) as T;
+  }
+
+  if (obj && obj.constructor === Object) {
+    return Object.keys(obj).reduce((acc, key) => {
+      const camelCaseKey = toCamelCase(key);
+      // biome-ignore lint/suspicious/noExplicitAny: <explanation>
+      (acc as any)[camelCaseKey] = convertSnakeToCamelCase<T>(
+        // biome-ignore lint/suspicious/noExplicitAny: <explanation>
+        (obj as any)[key],
+      );
+      return acc;
+    }, {} as T);
+  }
+  return obj;
+}
+
+function toCamelCase(str: string): string {
+  return str.replace(/_([a-z])/g, (_, letter) => letter.toUpperCase());
+}
```

Comment:
Nice! Only comment is that if we do it this way, we can remove the `any` and also have nice types outputted from the result!

So you'll get this:
const input = { snake_case: 1 }
const output = convertSnakeToCamelCase(input)

`output` will be of type: `{ snakeCase: 1 }`

```suggestion
/**
 * Converts snake_case keys to camelCase in an object or array of objects.
 * @param {T} obj - The object, array, or string to convert. (required)
 * @returns {T} The converted object, array, or string.
 */
export function convertSnakeToCamelCase<T>(obj: T): T {
  if (typeof obj === 'string') {
    return obj as T;
  }

  if (Array.isArray(obj)) {
    return obj.map((item) => convertSnakeToCamelCase(item)) as T;
  }

  if (obj && obj.constructor === Object) {
    return Object.keys(obj).reduce((acc, key) => {
      const camelCaseKey = toCamelCase(key);
      // biome-ignore lint/suspicious/noExplicitAny: <explanation>
      (acc as any)[camelCaseKey] = convertSnakeToCamelCase<T>(
        // biome-ignore lint/suspicious/noExplicitAny: <explanation>
        (obj as any)[key],
      );
      return acc;
    }, {} as T);
  }
  return obj;
}

function toCamelCase(str: string): string {
  return str.replace(/_([a-z])/g, (_, letter) => letter.toUpperCase());
}
```

Reply:
Done